docs/cue.md

```diff
@@ -60,5 +78,89 @@ it should contain:
 - under `spec.plugin` :
   - the query's `kind`.
   - the query's `spec` containing:
-		- a `datasource` field that holds the `kind` of datasource corresponding to this query type.
-		- any other field you want for this query plugin.
+    - a `datasource` field that holds the `kind` of datasource corresponding to this query type.
+    - any other field you want for this query plugin.
+
+# Migration from Grafana
+
+A Perses plugin can optionally embed a `mig.cuepart` file that is basically describing in CUE language how to convert a given Grafana object into an instance of this plugin. In such case your plugin is considered as the Perses equivalent of this Grafana object type, and it will be used as part of the translation process when a Grafana dashboard is received on the `/migrate` endpoint.
+
+## Variable
+
+A variable migration file looks like the following:
+
+```cue
+if #var.type == "custom" || #var.type == "interval" {
+    kind: "StaticListVariable"
+    spec: {
+        values: [ for _, option in #var.options if option.value !~ "^\\$__.*$" { // the if filters out the grafana global vars that'd be causing validation issues later (e.g "__auto_interval_sampling is used but not defined")
+            option.value
+        }]
+    }
+}
+```
+- The file has to be named `mig.cuepart`. 
```

Comment:
Yes that was just to push some kind of governance let's say, since we expect a single file with a specific extension, but indeed we can remove the constraint on the file name.